### PR TITLE
MDS-3009 New NoW document types and sub types

### PR DESCRIPTION
**Pull request by JustinMacaulay**
Adds a new table to hold NoW document sub types to allow for better filtering and grouping of documents on the admin tab of the NoW processing. Also adds a "Maps" section to the draft permit which will contain all documents of subtype "MDO".

### CODE REVIEW

File: services/core-web/src/components/Forms/noticeOfWork/EditNoticeOfWorkDocumentForm.js [hunk 1]
```diff
@@ -65,10 +65,22 @@ export class EditNoticeOfWorkDocumentForm extends Component {
...
+          return this.props.categoriesToShow.includes(subType);
+        }
+        if (this.props.categoriesToShow.length > 0) {
+          return this.props.categoriesToShow.includes(value);
+        }
+        return true;
+      }
     );
 
+    // .filter(
```
Comment (mmumby) [inline services/core-web/src/components/Forms/noticeOfWork/EditNoticeOfWorkDocumentForm.js:79]: delete

File: services/core-web/src/components/Forms/noticeOfWork/EditNoticeOfWorkDocumentForm.js [hunk 2]
```diff
@@ -65,10 +65,22 @@ export class EditNoticeOfWorkDocumentForm extends Component {
...
-        this.props.categoriesToShow.length > 0 ? this.props.categoriesToShow.includes(value) : value
+      ({ subType, value }) => {
+        if (subType && this.props.categoriesToShow.length > 0) {
+          return this.props.categoriesToShow.includes(subType);
+        }
+        if (this.props.categoriesToShow.length > 0) {
+          return this.props.categoriesToShow.includes(value);
+        }
+        return true;
+      }
```
Comment (mmumby) [inline services/core-web/src/components/Forms/noticeOfWork/EditNoticeOfWorkDocumentForm.js:76]: This needs to be an array not a boolean - so if there is nothing to filter, return the entire set

File: services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js [hunk 3]
```diff
@@ -91,21 +95,22 @@ export const NOWDocuments = (props) => {
...
+    const filtered = noticeOfWorkApplicationDocumentTypeOptions.filter(({ subType, value }) => {
+      if (subType && categoriesToShow.length > 0) {
+        return categoriesToShow.includes(subType);
+      }
+      if (categoriesToShow.length > 0) {
+        return categoriesToShow.includes(value);
+      }
+      return true;
+    });
+
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js:108]: same as here

File: services/core-web/src/components/Forms/noticeOfWork/EditNoticeOfWorkDocumentForm.js (hunk 1 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/Forms/noticeOfWork/EditNoticeOfWorkDocumentForm.js:79]: removed

**Review by mhuseinov — APPROVED** [review]

**Pull request merged**